### PR TITLE
Add kustomization of manifests

### PR DESCRIPTION
**Pull request by jichenjc**
-created vm/bm model folder
-patch manifest to BM deployment so it should reject VM env
-I didn't change major manifest usage (as I don't know whether we have other usage other than CI so don't want to break them), that's use a symlink to keep original things still working
-updated the build manifest so new manifest file will be generated in _output folder

Signed-off-by: jichenjc <jichenjc@cn.ibm.com>

### CODE REVIEW

**Comment by ruomengh** [comment]
> @jichenjc looks cool! Can you also update [README](https://github.com/sustainable-computing-io/kepler/blob/main/README.md) and [installation guide](https://github.com/sustainable-computing-io/kepler/tree/main/manifests#installing-kepler-on-kubernetes)?

+1. It would be helpful to update installation guide as well.

**Review by marceloamaral — CHANGES_REQUESTED** [review]
Seems pretty good!

Can you also update the [cluster-deploy.sh](https://github.com/sustainable-computing-io/kepler/blob/main/hack/cluster-deploy.sh#L25) file?
This PR changes the generated manifest deployment folder so we need to update it

We could also include an environment variable to select the correct manifest, like:
`BARE_METAL_NODE=True`

We make it True by default

**Comment by marceloamaral** [comment]
Or `BARE_METAL_NODE_ONLY=True` since there are scenarios that mix them

File: manifests/README.md [hunk 1]
```diff
@@ -10,6 +10,10 @@ The operating system must provide:
 Consult the documentation of your Linux distribution on details for enabling these prerequisites.
 ## Installing Kepler on Kubernetes
 
+Simple option:
+
+For people who just want a quick try:
+
 ```bash
 # NOTE: These manifest take care of creating the namespace and Kepler exporter
 kubectl apply -f kubernetes/deployment.yaml
```
Comment (rootfs) [inline manifests/README.md:19]: is this path still valid?

Comment (jichenjc) [reply manifests/README.md:19]: technicaly yes, as there is a symlink to base (deployment.yaml -> base/deployment.yaml) and  base/deployment.yaml
still useable for now

the reason for doing this is to make things simple(so first time user has less step to perform), but I think sooner or later we might need run kustomize inevitable

**Comment by jichenjc** [comment]
@rootfs @marceloamaral can you help take another look? Thanks

**Review by rootfs — APPROVED** [review]

**Comment by rootfs** [comment]
lgtm

**Review by marceloamaral — APPROVED** [review]
Looks great, thanks!
/lgtm

**Pull request merged**